src/main/java/org/prebid/server/handler/AuctionHandler.java

```diff
@@ -167,15 +183,54 @@ private PreBidRequestContext updateAppAndNoCookieAndImpsRequestedMetrics(
                 .collect(Collectors.toList());
     }
 
+    private Future<Map<Integer, Boolean>> resolveVendorsToGdpr(PreBidRequestContext preBidRequestContext,
+                                                               List<AdapterResponse> adapterResponses) {
+        final Set<Integer> vendorIds = adapterResponses.stream()
+                .map(adapterResponse -> adapterResponse.getBidderStatus().getBidder())
+                .filter(bidderCatalog::isActive)
+                .map(bidder -> bidderCatalog.metaInfoByName(bidder).info().getGdpr().getVendorId())
+                .collect(Collectors.toSet());
+
+        final boolean hostVendorIdIsMissing = gdprHostVendorId != null && !vendorIds.contains(gdprHostVendorId);
+        if (hostVendorIdIsMissing) {
+            vendorIds.add(gdprHostVendorId);
+        }
+
+        final String gdpr = GdprUtils.gdprFrom(preBidRequestContext.getPreBidRequest().getRegs());
+        final String gdprConsent = GdprUtils.gdprConsentFrom(preBidRequestContext.getPreBidRequest().getUser());
+        final String ip = useGeoLocation ? preBidRequestContext.getIp() : null;
+
+        return gdprService.resultByVendor(GDPR_PURPOSES, vendorIds, gdpr, gdprConsent, ip,
+                preBidRequestContext.getTimeout())
+                .map(gdprResponse -> toVendorsToGdpr(gdprResponse.getVendorsToGdpr(), hostVendorIdIsMissing));
+    }
+
+    private Map<Integer, Boolean> toVendorsToGdpr(Map<Integer, Boolean> vendorsToGdpr, boolean hostVendorIdIsMissing) {
+        final Map<Integer, Boolean> result;
+
+        if (Objects.equals(vendorsToGdpr.get(gdprHostVendorId), false)) {
+            result = Collections.emptyMap(); // deny all by host vendor
+        } else if (hostVendorIdIsMissing) {
+            final Map<Integer, Boolean> vendorsToGdprWithoutHost = new HashMap<>(vendorsToGdpr);
+            vendorsToGdpr.remove(gdprHostVendorId); // just to be clean with bidders
```

Comment:
Seems gdprHostVendorId should be removed from vendorsToGdprWithoutHost map. Add test for this case